src/test/java/com/microsoft/sqlserver/jdbc/requestboundary/RequestBoundaryMethodsTest.java

```diff
@@ -0,0 +1,421 @@
+package com.microsoft.sqlserver.jdbc.requestboundary;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.sql.CallableStatement;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+import org.junit.jupiter.api.Test;
+import org.junit.platform.runner.JUnitPlatform;
+import org.junit.runner.RunWith;
+
+import com.microsoft.sqlserver.jdbc.SQLServerConnection;
+import com.microsoft.sqlserver.testframework.AbstractSQLGenerator;
+import com.microsoft.sqlserver.testframework.AbstractTest;
+import com.microsoft.sqlserver.testframework.PrepUtil;
+import com.microsoft.sqlserver.testframework.Utils;
+import com.microsoft.sqlserver.testframework.util.RandomUtil;
+
+/**
+ * A class for testing Request Boundary Methods.
+ */
+@RunWith(JUnitPlatform.class)
+public class RequestBoundaryMethodsTest extends AbstractTest {
+    SQLServerConnection con = null;
+    Statement stmt = null;
+    PreparedStatement pstmt = null;
+    CallableStatement cstmt = null;
+    ResultSet rs = null;
+
+    /**
+     * Tests Request Boundary methods with SQLServerConnection properties that are modifiable through public APIs.
+     * 
+     * @throws SQLException
+     */
+
+    @Test
+    public void testModifiableConnectionProperties() throws SQLException {
+        // List of SQLServerConnection fields that can be modified through public APIs.
+        boolean autoCommitMode1 = true;
+        int transactionIsolationLevel1 = SQLServerConnection.TRANSACTION_READ_COMMITTED;
+        int networkTimeout1 = 5000;
+        int holdability1 = ResultSet.HOLD_CURSORS_OVER_COMMIT;
+        boolean sendTimeAsDatetime1 = true;
+        int statementPoolingCacheSize1 = 0;
+        boolean disableStatementPooling1 = true;
+        int serverPreparedStatementDiscardThreshold1 = 10;
+        boolean enablePrepareOnFirstPreparedStatementCall1 = false;
+        String sCatalog1 = "model";
+
+        boolean autoCommitMode2 = false;
+        int transactionIsolationLevel2 = SQLServerConnection.TRANSACTION_SERIALIZABLE;
+        int networkTimeout2 = 10000;
+        int holdability2 = ResultSet.CLOSE_CURSORS_AT_COMMIT;
+        boolean sendTimeAsDatetime2 = false;
+        int statementPoolingCacheSize2 = 10;
+        boolean disableStatementPooling2 = false;
+        int serverPreparedStatementDiscardThreshold2 = 100;
+        boolean enablePrepareOnFirstPreparedStatementCall2 = true;
+        String sCatalog2 = "tempdb";

```

Comment:
Why are we hardcoding database names (model & tempdb) here? This test would fail if attempted to run with Azure DB specified in connection String.

Reply:
Agreed.